test/routes/api/user/user.test.ts

```diff
@@ -0,0 +1,159 @@
+import { it, describe, beforeEach, afterEach, before } from 'node:test'
+import assert from 'node:assert'
+import { build } from '../../../helper.js'
+import { FastifyInstance } from 'fastify'
+
+async function createUser (app: FastifyInstance, userData: Partial<{ username: string; password: string }>) {
+  const [id] = await app.knex('users').insert(userData)
+  return id
+}
+
+async function deleteUser (app: FastifyInstance, username: string) {
+  await app.knex('users').delete().where({ username })
+}
+
+async function updatePasswordWithLoginInjection (app: FastifyInstance, username: string, payload: { currentPassword: string; newPassword: string }) {
+  return await app.injectWithLogin(username, {
+    method: 'PUT',
+    url: '/api/user/update-password',
+    payload
+  })
+}
+
+describe('User API', () => {
+  // Hashed value of `Password123$`
+  const Password123$ = 'ff57faf149a2bcab41bf7ecbbc8ce491.3ce6b34ea3edb3f0a09f811440885bfeda612832c04bfddc9d4b906019d97fa0'
+  let app: FastifyInstance
+
+  before(async () => {
+    const app = await build()
+    const users = ['random-user-0', 'random-user-1', 'random-user-2', 'random-user-3', 'random-user-4', 'random-user-5']
+
+    for (const user of users) {
+      await createUser(app, {
+        username: user,
+        password: Password123$
+      })
+    }
```

Comment:
I don't understand why you create all the users here, than recreate them individually in all the tests.

Reply:
it's an oversight